src/admin/class-options-form-generator.php

```diff
@@ -0,0 +1,301 @@
+<?php
+/**
+ * Duplicate Post plugin file.
+ *
+ * @package Yoast\WP\Duplicate_Post\Admin
+ */
+
+namespace Yoast\WP\Duplicate_Post\Admin;
+
+use Yoast\WP\Duplicate_Post\Utils;
+
+/**
+ * Class Options_Form_Generator
+ *
+ * @package Yoast\WP\Duplicate_Post\Admin
+ */
+class Options_Form_Generator {
+
+	/**
+	 * The Options_Inputs instance.
+	 *
+	 * @var Options_Inputs
+	 */
+	protected $inputs;
+
+	/**
+	 * Options_Form_Generator constructor.
+	 *
+	 * @param Options_Inputs $inputs The Options_Inputs instance.
+	 */
+	public function __construct( Options_Inputs $inputs ) {
+		$this->inputs = $inputs;
+	}
+
+	/**
+	 * Generates the HTML output of an input control, based on the passed options.
+	 *
+	 * @param array  $options       The options to base the input on.
+	 * @param string $parent_option The parent option, used for grouped inputs. Optional.
+	 *
+	 * @return string The HTML output.
+	 */
+	public function generate_options_input( array $options, $parent_option = '' ) {
+		$output = '';
+
+		foreach ( $options as $option => $option_values ) {
+			// Skip empty options.
+			if ( empty( $option_values ) ) {
+				continue;
+			}
+
+			// Check for support of the current WordPress version.
+			if ( array_key_exists( 'version', $option_values ) && version_compare( \get_bloginfo( 'version' ), $option_values['version'] ) < 0 ) {
+				continue;
+			}
+
+			if ( array_key_exists( 'sub_options', $option_values ) ) {
+				$output .= $this->generate_options_input( $option_values['sub_options'], $option );
+
+				continue;
+			}
+
+			// If callback, call it.
+			if ( array_key_exists( 'callback', $option_values ) ) {
+				$output .= $this->{$option_values['callback']}();
+
```

Comment:
Remove whiteline for consistency

Reply:
It'd be less readable after removing this whiteline, as there's more than one thing happening in this if-statement, whereas the previous one only has `continue`.

Reply:
Fair point, agreed 👍🏻